azurerm/internal/services/mssql/resource_arm_mssql_virtual_machine.go

```diff
@@ -0,0 +1,503 @@
+							Type:         schema.TypeString,
+							Optional:     true,
+							Sensitive:    true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+
+						"sql_connectivity_update_username": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Sensitive:    true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+					},
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmMsSqlVirtualMachineCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MSSQL.SQLVirtualMachinesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	vmId := d.Get("virtual_machine_id").(string)
+	id, err := compute.ParseVirtualMachineID(vmId)
+	if err != nil {
+		return err
+	}
+	name := id.Name
+	resourceGroupName := id.ResourceGroup
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroupName, name, "*")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for present of existing Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q): %+v", name, resourceGroupName, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_mssql_virtual_machine", *existing.ID)
+		}
+	}
+
+	// get location from vm
+	vmclient := meta.(*clients.Client).Compute.VMClient
+	respvm, err := vmclient.Get(ctx, resourceGroupName, name, "")
+	if err != nil {
+		return fmt.Errorf("Error making Read request on Azure Virtual Machine %s: %+v", name, err)
+	}
+
+	if *respvm.Location == "" {
+		return fmt.Errorf("Error location is empty from making Read request on Azure Virtual Machine %s: %+v", name, err)
+	}
+
+	properties := sqlvirtualmachine.Properties{
+		VirtualMachineResourceID:               utils.String(d.Get("virtual_machine_id").(string)),
+		SQLServerLicenseType:                   sqlvirtualmachine.SQLServerLicenseType(d.Get("sql_license_type").(string)),
+		SQLManagement:                          sqlvirtualmachine.Full,
+		AutoPatchingSettings:                   expandArmSqlVirtualMachineAutoPatchingSettings(d.Get("auto_patching").([]interface{})),
+		KeyVaultCredentialSettings:             expandArmSqlVirtualMachineKeyVaultCredential(d.Get("key_vault_credential").([]interface{})),
+		ServerConfigurationsManagementSettings: expandArmSqlVirtualMachineServerConfigurationsManagement(d.Get("server_configuration").([]interface{})),
+	}
+
+	if sqlVirtualMachineGroupResourceID, ok := d.GetOk("sql_virtual_machine_group_id"); ok {
+		properties.SQLVirtualMachineGroupResourceID = utils.String(sqlVirtualMachineGroupResourceID.(string))
+	}
+
+	parameters := sqlvirtualmachine.SQLVirtualMachine{
+		Location:   utils.String(*respvm.Location),
+		Properties: &properties,
+		Tags:       tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
```

Comment:
could we just combine these
```suggestion
	parameters := sqlvirtualmachine.SQLVirtualMachine{
		Location:   utils.String(*respvm.Location),
		Properties: &sqlvirtualmachine.Properties{
			VirtualMachineResourceID:               utils.String(d.Get("virtual_machine_id").(string)),
			SQLServerLicenseType:                   sqlvirtualmachine.SQLServerLicenseType(d.Get("sql_license_type").(string)),
			SQLManagement:                          sqlvirtualmachine.Full,
			AutoPatchingSettings:                   expandArmSqlVirtualMachineAutoPatchingSettings(d.Get("auto_patching").([]interface{})),
			KeyVaultCredentialSettings:             expandArmSqlVirtualMachineKeyVaultCredential(d.Get("key_vault_credential").([]interface{})),
			ServerConfigurationsManagementSettings: expandArmSqlVirtualMachineServerConfigurationsManagement(d.Get("server_configuration").([]interface{})),
		},
		Tags:       tags.Expand(d.Get("tags").(map[string]interface{})),
	}
```